# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ihaseeb0081/haseeb-repository-ML/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I selected Lane 2:
 *Refresh / Content Opportunity* Scoring because I am interested in how data can be used to improve content performance. Large websites generate a significant amount of information, making manual analysis time-consuming and inefficient. This lane allows me to explore how machine learning can identify patterns in page performance and help prioritize content review opportunities. It combines real-world decision making with practical machine learning concepts.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

I want to improve the decision of which content pages should be reviewed first for refresh or improvement. The main users of this work would be content and SEO teams, who can use the ranked recommendations to prioritize their limited time and resources. A wrong recommendation could cause the team to spend effort on pages with little improvement potential, while missing pages that genuinely need attention. This could lead to wasted resources or missed opportunities to improve content performance and visibility.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [17]:
!find /content -name "*.csv"

/content/haseeb-repository-ML/data/raw/content_refresh_anonymized.csv
/content/haseeb-repository-ML/outputs/refresh_queue_sample.csv
/content/sample_data/mnist_train_small.csv
/content/sample_data/california_housing_train.csv
/content/sample_data/mnist_test.csv
/content/sample_data/california_housing_test.csv


In [18]:
import pandas as pd

df = pd.read_csv("/content/haseeb-repository-ML/data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Average impressions (90d):", round(df["impressions_90d"].mean(), 2))
print("Pages trending down:", (df["trend_direction"] == "down").sum())
print("Pages trending down (%):", round((df["trend_direction"] == "down").mean() * 100, 2))

Total pages: 30000
Average impressions (90d): 5200.37
Pages trending down: 16262
Pages trending down (%): 54.21


The starter dataset contains 30,000 content pages, providing a substantial amount of data for analysis. The average number of impressions over the last 90 days is 5,200.37, showing that many pages receive meaningful search visibility. Additionally, 16,262 pages (54.21% of the dataset) are currently trending downward. These numbers suggest that content refresh prioritization is a relevant business problem, as a large portion of pages may benefit from review or improvement. This makes Lane 2 (Refresh / Content Opportunity Scoring) a valuable direction for further investigation.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project can identify observed patterns in content performance and provide decision-support recommendations for which pages may deserve review. It can help prioritize pages based on measurable signals in the dataset. However, it cannot prove that refreshing a page will definitely improve rankings or traffic, nor can it predict future search engine behavior with certainty. The results should be treated as guidance rather than guaranteed outcomes.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.